slam-libraries/viam-cartographer/src/slam_service/slam_service.h

```diff
@@ -0,0 +1,80 @@
+#ifndef SLAM_SERVICE_H_
+#define SLAM_SERVICE_H_
+
+#include <grpc/grpc.h>
+#include <grpcpp/server.h>
+#include <grpcpp/server_context.h>
+
+#include <atomic>
+#include <string>
+
+#include "../io/draw_trajectories.h"
+#include "../io/read_PCD_file.h"
+#include "../io/submap_painter.h"
+#include "cartographer/mapping/map_builder.h"
+#include "common/v1/common.grpc.pb.h"
+#include "common/v1/common.pb.h"
+#include "service/slam/v1/slam.grpc.pb.h"
+#include "service/slam/v1/slam.pb.h"
+
+using grpc::ServerContext;
+using viam::service::slam::v1::GetMapRequest;
+using viam::service::slam::v1::GetMapResponse;
+using viam::service::slam::v1::GetPositionRequest;
+using viam::service::slam::v1::GetPositionResponse;
+using viam::service::slam::v1::SLAMService;
+
+namespace viam {
+
+static const int checkForShutdownIntervalMicroseconds = 1e5;
+static std::atomic<bool> b_continue_session{true};
+
+using SensorId = cartographer::mapping::TrajectoryBuilderInterface::SensorId;
+const SensorId kRangeSensorId{SensorId::SensorType::RANGE, "range"};
+const SensorId kIMUSensorId{SensorId::SensorType::IMU, "imu"};
+
+class SLAMServiceImpl final : public SLAMService::Service {
+   public:
+    // GetPosition returns the relative position of the robot w.r.t the "origin"
+    // of the map, which is the starting point from where the map was initially
+    // created.
+    ::grpc::Status GetPosition(ServerContext *context,
+                               const GetPositionRequest *request,
+                               GetPositionResponse *response) override;
+
+    // GetMap returns either an image or a pointcloud, depending on the MIME
+    // type requested
+    ::grpc::Status GetMap(ServerContext *context, const GetMapRequest *request,
+                          GetMapResponse *response) override;
+
+    void CreateMap();
```

Comment:
What's the reason to include this function? I think we could instead leave it out, since we won't need to have it when we implement ProcessDataOnline/Offline.

Reply:
I put this here for two reasons:
1. To be able to run mapping from within the slam service already now by calling this function as part of the service - so for basic testing purposes while we continue developing the slam service
2. As a first & intermediate step for extracting code from `main_old.cc` into the slam service functions. I was hoping this would make it easier for folks picking up the next cartographer tickets to see how mapping + png map creation + pbstream map saving is started & executed. Those could then be further picked apart into the respective functions.

Would it help if I added a comment describing that above the function? That it is a temporary placeholder until the other slam service functions + ProcessDataOnline/Offline are implemented?